### DISCUSSION TITLE
Generalized modal analysis

### DISCUSSION THREAD

**Discussion by ckesanapalli** [capytaine/capytaine#78] [Ideas]
I recently came across the capytaine library and it is really easy to use and understand. 

Can capytaine perform hydroelasticity using Generalized modal analysis? If not, are you planning to implement it anytime soon?

Thank you
Chaitanya Kesanapalli

**Comment by mancellin** [comment]
Thank you for the kind words.

Yes, you can define arbitrary deformation modes. There is a code sample in the user documentation: look for the "Custom degree of freedom" section in the cookbook.

**Reply by vaibhavraghavan** [reply]
Hi Matthieu,

I was wondering how the custom degrees of freedom work. For example, if I wanted to model an OWC with the internal free surface, then I can use a custom mode for the free surface. I was wondering whether this would work the same way as seen in WAMIT with the thin disk approach?

Regards,

Vaibhav

**Reply by mancellin** [reply]
@vaibhavraghavan Thank you for your message. 
There is kind of two questions here:
- Can you represent the internal free surface as some kind of flexible membrane and compute with Capytaine the effect of custom deformation modes? Yes, I guess it is possible.
- Does it make sense for the modelling of OWC? I have no idea.
